dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java

```diff
@@ -398,13 +400,19 @@ private void processQuery(String filter, String sortCol, String sortOrder, Crite
                     query.orderBy(cb.asc(cb.size(entry.<Collection>get("starredUsers"))), cb.desc(entry.get("id")));
                 }
             } else {
-                Path<Object> sortPath = entry.get(sortCol);
-                if (!Strings.isNullOrEmpty(sortOrder) && "desc".equalsIgnoreCase(sortOrder)) {
-                    query.orderBy(cb.desc(sortPath), cb.desc(entry.get("id")));
-                    predicates.add(sortPath.isNotNull());
-                } else {
-                    query.orderBy(cb.asc(sortPath), cb.desc(entry.get("id")));
-                    predicates.add(sortPath.isNotNull());
+                try {
+                    Path<Object> sortPath = entry.get(sortCol);
+                    if (!Strings.isNullOrEmpty(sortOrder) && "desc".equalsIgnoreCase(sortOrder)) {
+                        query.orderBy(cb.desc(sortPath), cb.desc(entry.get("id")));
+                        predicates.add(sortPath.isNotNull());
+                    } else {
+                        query.orderBy(cb.asc(sortPath), cb.desc(entry.get("id")));
+                        predicates.add(sortPath.isNotNull());
+                    }
+                } catch (IllegalArgumentException e) {
```

Comment:
Generally, if possible, if there's a possibility that the argument to a functions is invalid, it's better to check and take the appropriate action (logging, throwing, etc) before the function call.  The above will `catch` any `IllegalArgumentException`s thrown in the block, which is probably not what we're after, and if the behavior of `entry.get` changes, this code might break.

Reply:
changed to check before function call now